BotController/BotController.cpp

```diff
@@ -1,12 +1,340 @@
-// BotController.cpp : This file contains the 'main' function. Program execution begins and ends there.
-//
-
 #include <iostream>
+#include <Winuser.h>
 #include "Robot.hpp"
+//include "RGBQuadValues.hpp"
 
 int main()
 {
-    std::cout << "Hello World!\n";
+    Robot robot = Robot();
+
+    //how exact the pixel color's RGBs need to be
+    double tolerance = 0.0;
```

Comment:
Consider `constexpr` to improve stack utilization and improve clarity. If the tolerance changes, we'll need to leave it as mutable, but otherwise it should be `const`, or even better, `constexpr`.
```suggestion
constexpr double tolerance = 0.0;
```

Reply:
I'd be careful about 0 tolerances because floating point round off can be really problematic. I recommend using C++ small floats like:
```suggestion
    constexpr double tolerance = 1e-6; // 1 x 10^-6
```